icd/api/pipeline_compiler.cpp

```diff
@@ -763,23 +897,28 @@ VkResult PipelineCompiler::CreateGraphicsPipelineBinary(
         }
     }
 
-    if ((pPipelineBinaryCache != nullptr) &&
-        (isUserCacheHit == false) &&
-        (result == VK_SUCCESS))
+#if XGL_USE_EXPERIMENTAL_SHADER_CACHE_PIPELINES
+    if ((pPipelineCache != nullptr) &&
+        (result == VK_SUCCESS) &&
+        (cacheResult == Util::Result::NotFound))
     {
-        cacheResult = pPipelineBinaryCache->StorePipelineBinary(
-            pCacheId,
-            *pPipelineBinarySize,
-            *ppPipelineBinary);
+        Llpc::IShaderCache* const pLlpcShaderCache =
+            pPipelineCache->GetShaderCache(deviceIdx).GetCachePtr().pLlpcShaderCache;
 
-        VK_ASSERT(Util::IsErrorResult(cacheResult) == false);
+        if (cacheType == PipelineCompilerTypeLlpc && pLlpcShaderCache != nullptr)
+        {
+            cacheResult = StorePipelineBinary(
+                pCacheId, pPipelineBinarySize, ppPipelineBinary, pLlpcShaderCache);
+        }
     }
 
-    if ((m_pBinaryCache != nullptr) &&
-        (isInternalCacheHit == false) &&
+#else // XGL_USE_EXPERIMENTAL_SHADER_CACHE_PIPELINES
+
+    if ((pPipelineBinaryCache != nullptr) &&
+        (isUserCacheHit == false) &&
         (result == VK_SUCCESS))
     {
-        cacheResult = m_pBinaryCache->StorePipelineBinary(
```

Comment:
if ((m_pBinaryCache != nullptr) &&
        (isInternalCacheHit == false) &&
        (result == VK_SUCCESS))
    {
        cacheResult = m_pBinaryCache->StorePipelineBinary(
            pCacheId,
            *pPipelineBinarySize,
            *ppPipelineBinary);

        VK_ASSERT(Util::IsErrorResult(cacheResult) == false);
}
this piece code exists in original code path, it's gone, do you delete it intentionally? without the piece of code, out internal tests are failed.

Reply:
This was an unintentional change made when porting the code. Thank for catching it!

Reply:
Also removed unintended changes in the if condition.